### PR TITLE
Fix parameter error parsing in taxcalcio module

### LINKED ISSUE #2622: Handling ParamTools error messages
@hdoupe From the errors reported in Issue #2621, it looks like there has been change in the `paramtools` errors from being strings to dictionaries and therefore an error is occurring on line 282 of `taxcalcio.py` and users don't see the intended validation error message.  No current `taxcalc` unit tests seem to catch this.  Can  you confirm that that this is the case and perhaps propose a fix (and ideally a test to catch this type of thing going forward)?

### PR DESCRIPTION
**Pull request by hdoupe**
Resolves #2622.

### CODE REVIEW

File: taxcalc/parameters.py [hunk 1]
```diff
@@ -689,11 +689,11 @@ def num_years(self):
 
     @property
     def parameter_warnings(self):
-        return self.errors or ""
+        return self.errors or {}
```
Comment (hdoupe) [inline taxcalc/parameters.py:692]: The parameter errors are a dictionary that looks like this:

```python
{'AMEDT_rt': ['AMEDT_rt[year=2021] 1.8 > max 1 ']}
```

This change sets the default value to be a dictionary to keep the type consistent.

**Comment by jdebacker** [comment]
@hdoupe Looks very good and I was able to run all tests locally, plus get the proper error message from the json file given in Issue #2621.

But before merging, can you get that new line of code covered?  I think it just involves passing a reform file to the baseline argument in the new test you wrote, which does catch all the new lines for the reform file error messages.

Thanks for your work on this!

**Comment by hdoupe** [comment]
@jdebacker Looks like the code coverage test is passing now! Thanks for reviewing this.

**Comment by jdebacker** [comment]
@hdoupe Thanks for getting that line covered -- and for the PR.  Merging.

**Pull request merged**